awx/api/serializers.py

```diff
@@ -4980,9 +4983,18 @@ def validate_node_state(self, value):
         return value
 
     def validate_hostname(self, value):
+        """
+        - Hostname cannot be "localhost" - but can be something like localhost.domain
+        - Cannot change the hostname of an-already instantiated & initialized Instance object
+        """
         if self.instance and self.instance.hostname != value:
             raise serializers.ValidationError("Cannot change hostname.")
 
+        HostnameRegexValidator(regex="[a-z0-9]([-a-z0-9]*[a-z0-9])?", flags=re.IGNORECASE)(value)
+        HostnameRegexValidator(regex="^localhost$|^127.0.0.1$", flags=re.IGNORECASE, inverse_match=True, message="hostname cannot be localhost or 127.0.0.1")(
```

Comment:
In my opinion we also need to exclude the IPv6 localhost.

Reply:
Also, I suspect that we can get by with just a plain `RegexValidator` for these (and really just to handle the IPv6 localhost case).

Reply:
You can compress ipv6 localhost by replacing any combinations of the zeros with empty strings, so you can have a bunch of colons and `1`, or various combinations thereof.